nodeadm/api/v1alpha1/nodeconfig_types.go

```diff
@@ -79,6 +79,7 @@ type ContainerdOptions struct {
 // InstanceOptions determines how the node's operating system and devices are configured.
 type InstanceOptions struct {
 	LocalStorage LocalStorageOptions `json:"localStorage,omitempty"`
+	NodeName     NodeNameOptions     `json:"nodeName,omitempty"`
```

Comment:
I think we should avoid making this so strongly-typed. We don't want this to be an option in the long term, we just need a way to opt-in to the behavior before it becomes the *only* behavior on AL2025.

Something like @ndbaker1 's original feature flags approach makes more sense to me, because those values are not codified in the schema.